src/fsharp/LanguageFeatures.fs

```diff
@@ -19,13 +19,14 @@ open System
 /// LanguageFeature enumeration
 [<RequireQualifiedAccess>]
 type LanguageFeature =
-    | LanguageVersion46 = 0
-    | LanguageVersion47 = 1
-    | SingleUnderscorePattern = 2
-    | WildCardInForLoop = 3
-    | RelaxWhitespace = 4
-    | NameOf = 5
-    | ImplicitYield = 6
+    | PreviewVersion = 0
```

Comment:
Please don't change these numbers in future.

And I still believe it'd be better to have separate enums for language versions and features.

Reply:
@auduchinok.
They aren't public and shouldn't be relied on.  In fact, I wouldn't mind deleting the individual feature ID's to remove the possibility that we try to add a --enablefeature:WildCardInForLoop  option.

Reply:
It's probably reasonable to think they may become public in FCS at some point when tooling wants to see which features are available.

Reply:
@auduchinok,

The smallest subdivision of the language is language version, from it one can infer the set of features available.  If we choose to add or I suppose remove features, then we will release a new language version, similarly to how we release new nuget packages and assembly versions for FSharp.Core.

It is not our intention to make the information in these structures public.  I can see a good argument for removing
````
 | Preview = 0
 | LanguageVersion46 = 1
 | LanguageVersion47 = 2
````
from the LanguageFeature enumeration, given that it has so far not proven to be necessary to do a SupportsFeature.LanguageVersion4.7 call.

Anyway, my 2 cents, and probably not even worth that amount of coin.

Reply:
> The smallest subdivision of the language is language version, from it one can infer the set of features available.

@KevinRansom Thanks for your reply. I don't, however, realise how it should be inferred by external tools using FCS given the whole module is currently internal (unless the whole logic is reimplemented in each tool). There may be situations where tooling behaviour can also change depending on whether some feature is supported in particular language version.

Reply:
@auduchinok Which situations? The way it works in Roslyn, which is the model we're adopting, is that tools assume `latest`. But because diagnostics will kick in if something doesn't actually compile, the only weird thing is displaying information when you're in an invalid state.

Here's an example in C# 7.3 but using nullable:

![image](https://user-images.githubusercontent.com/6309070/61162502-bf7e2400-a4bd-11e9-938c-42c7674d32cc.png)

I could believe two opposing arguments:

1. That it is incorrect behavior to show `string?`, since it's not allowed in the first place.
2. That it is correct to show `string?`, since it is what the programmer wrote

Since this can be resolved either by turning `string?` to `string` or setting LangVersion ot 8.0 (or preview), it's unclear to me which is the correct behavior. But it's certainly easier from a tooling standpoint to not flow LangVersion into every tooling feature.

Reply:
@cartermp I don't have a particular example in mind but I can definitely imagine situations where an IDE could, for instance, produce different generated code or do code formatting differently depending on the language version in future.